lib/libzfs/libzfs_dataset.c

```diff
@@ -3854,6 +3858,14 @@ zfs_create(libzfs_handle_t *hdl, const char *path, zfs_type_t type,
 			    "size"));
 			return (zfs_error(hdl, EZFS_BADPROP, errbuf));
 		}
+
+		if ((ashift > 0ULL) &&
+		    (blocksize < (1ULL << ashift))) {
+			(void) fprintf(stderr, dgettext(TEXT_DOMAIN, "Warning: "
+			    "volblocksize (%llu) < ashift (%llu / %llu)\n"
+			    "means all writes are amplified and space wasted.\n"),
+			    blocksize, ashift, 1ULL << ashift);
```

Comment:
We've been explicitly adding a `u_longlong_t` cast to resolve the compiler warnings for the `%llu` types.  Alternately, you could use `zfs_nicebytes` to pretty-print some of values with human readable suffixes.  Either solution seems fine to me.